_drafts/javascript.md

```diff
@@ -0,0 +1,463 @@
+---
+state: In Review
+---
+
+# JavaScript
+
+## Motivation
+
+This evolution proposal is part of point 3 of [the Hubot 3.0 milestone](https://github.com/hubotio/evolution/tree/01dd8577c8e616921c3c6fb61b3ccc3801cdaf1f#hubot-30):
+
+> Modernize the project by translating CoffeeScript to JavaScript, improving integration with various developer tools, and adding features that make it easier for developers to automate their workflow.
+
+**Scope:**
+
+- Simplify contribution / maintenance
+- Removal of tooling / build steps
+- Align Node support with [Node’s LTS schedule](https://github.com/nodejs/LTS#lts-schedule1)
+- maintain backwards compatibility of user scripts written in CoffeeScript
+
+**Out of scope:**
+
+- no new features or breaking changes (beyond dropping support for Node <v4)
+- Introduce any other tooling besides linter
+- Any non-trivial refactoring of the existing code base
+
+## Proposed solution
+
+Converting of all CoffeeScript to JavaScript which can run natively on Node 4 ([current maintenance version until April 2018](https://github.com/nodejs/LTS#lts-schedule1)).
+
+Probably the biggest challenge is that CoffeeScript classes are incompatible
+with es2015 classes: they cannot extend from each other. It is something that
+[CoffeeScript v2 is addressing](http://coffeescript.org/v2/#coffeescript-2).
+
+The hubot eco system expects CoffeeScript-compatible "classes" being return in
+`require('hubot')`, even when 3rd party adapters have been implemented in
+JavaScript themselves. To address that problem, I suggest we use es2015 `class`
+declarations internally but wrap them in CoffeeScript compatible classes in
+`/index.js`. Additionally, we can introduce a new officially support export at
+`require('hubot/es2015')` which returns our es2015 class declarations, so that
+3rd party developers can use es2015 class declarations themselves and inhert
+from ours.
+
+## Detailed process
+
+The steps of the conversation will be
+
+- [x] Convert source files from CoffeeScript to JavaScript.
+- [x] Test with popular projects depending on Hubot
+- [x] Convert test files from CoffeeScript to JavaScript.
+- [x] update package.json
+- [x] add script for JavaScript linting
+- [ ] Release `hubot@3.0.0`
+  - [ ] Update hubot’s version range `peerDependencies` of default scripts
+  - [ ] Publish `hubot@3.0.0` with `--tag=next`
+  - [ ] Convert default scripts to JavaScrip, release as new breaking versions with `@next` tag
+  - [ ] Convert generator to JavaScript, release new breaking version with `@next` tag
+  - [ ] Test all `@next` releases
+  - [ ] Set `@latest` tag on all modules
+- [ ] Update documentation
```

Comment:
The documentation should be updated before the release. I want to make sure that the files that are packaged include updated documentation with how to use the Javascript support.